### ISSUE TITLE
Use Markdown to style props description in the PropsTable

### ISSUE THREAD

**Issue by mlegait** [storybookjs/storybook#8059]
**Is your feature request related to a problem? Please describe.**
With `addon-docs`, it's already possible to style the component description using markdown in the comment related to the component.
Would it be possible to do the same for props description?

**Describe the solution you'd like**
If I use the markdown syntax in the props comments, I would like to have the description styled accordingly in the PropsTable in the Docs tab.

**Describe alternatives you've considered**
Not styling it? ;)

**Are you able to assist bring the feature to reality?**
no

**Additional context**
Here is an example of comment that would be nice once styled:
```javascript
/**
 * The type of the date picker:
 *
 * * `simple` - Without calendar dropdown.
 * * `single` - With calendar dropdown and single date.
 * * `range` - With calendar dropdown and a date range.
 */
```

**Comment by shilman** [comment]
I believe this was fixed in https://github.com/storybookjs/storybook/releases/tag/v5.3.0-alpha.36

**Comment by iansan5653** [comment]
@shilman Basic markdown appears to be supported (ie inline code), but markdown-style linebreaks are not preserved and lists are not working at all. Here's an example of such a list:

![image](https://user-images.githubusercontent.com/2294248/71683223-eed06800-2d5f-11ea-94d4-fe7d590c1832.png)

That was generated from this (truncated) comment:
```ts
  /**
   * The type of value expected by the input. Other props on this element depend
   * on the value of this prop.
   *
   * *Important Note*: Unlike with normal `<input>` elements, setting this will
   * not validate the input contents. This is because in this project we use
   * Formik and Yup to validate fields at the form-level, not at the individual
   * input level. It is still very important to set this value properly for
   * accessibility and user experience.
   *
   * Possible values are:
   *  * `"number"` Any number not represented by a more specific type.
   *  * `"password"` A password.
   *  * `"email"` An email address.
   *  * `"tel"` A phone or fax number. Shows the phone number keypad on
   *      mobile keyboards.
   * ...
   */
```

Despite #8533 being closed "in favor of using markdown format", it appears that the only way to insert breaks is still the HTML form `<br>`.

**Comment by shilman** [comment]
@iansan5653 you need to add an extra linebreak between the "Possible values are" and the list, per https://github.com/probablyup/markdown-to-jsx/issues/217

As for the paragraph spacing, fixed in #9307. Here's what it looks like after the tweak:

<img width="660" alt="Addons___Docs___stories_-_Basic_⋅_Storybook" src="https://user-images.githubusercontent.com/488689/71706171-27367b80-2e1e-11ea-8b16-ec37b29b066e.png">

**Comment by shilman** [comment]
Jiminy cricket!! I just released https://github.com/storybookjs/storybook/releases/tag/v5.3.0-rc.8 containing PR #9307 that references this issue. Upgrade today to try it out!

You can find this prerelease on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.
